istioctl/pkg/validate/validate.go

```diff
@@ -276,25 +277,71 @@ func validateFiles(istioNamespace *string, defaultNamespace string, filenames []
 	var errs, err error
 	var reader io.ReadCloser
 	warningsByFilename := map[string]validation.Warning{}
-	for _, filename := range filenames {
-		if filename == "-" {
+
+	processFile := func(path string) {
+		if path == "-" {
 			reader = io.NopCloser(os.Stdin)
 		} else {
-			reader, err = os.Open(filename)
+			reader, err = os.Open(path)
+			if err != nil {
+				errs = multierror.Append(errs, fmt.Errorf("cannot read file %q: %v", path, err))
+				return
+			}
 		}
 		if err != nil {
```

Comment:
This entire outdented `if` conditional from 291-293 seems pointless/redundant now? Maybe I'm missing something.

When would it ever get hit?